### DISCUSSION TITLE
opendronemap/nodeodm

### DISCUSSION THREAD

**Discussion by Moekali84** [akash-network/community#355] [General]
Hello Everyone,

is there a way to run Webodm within the Akash Network? the Problem i am facing trying to run that is that WEBODM relies on Docker Compose to orchestrate the different containers within that application. I can run each container on its own which does not really help when trying to run an application that has dependencies in different containers.
https://hub.docker.com/r/opendronemap/nodeodm

**Comment by ghost** [comment]
Hi @Moekali84 :wave: 

I have not tried deploying webodm / nodeodm myself, but you can write SDL with multiple services just like you would normally do it with the Docker Compose.

https://docs.akash.network/readme/stack-definition-language#services

here is an example of the Ghost deployment and mysql DB in a single SDL file:
https://github.com/akash-network/awesome-akash/blob/master/ghost/deploy-mysql.yaml

The only caveat is that you can't share the same storage mountpoints between the services in Akash.

Let me know if you have any further questions.

**Reply by ghost** [reply]
I went ahead and looked at the https://github.com/OpenDroneMap/WebODM repo. It appears it is using bunch of wrappers and docker-compose executions for the config.
I believe it is possible to convert everything into Akash SDL, but it isn't super straightforward as it requires one be experienced with WebODM and docker / the way shell wrappers & docker-compose are used in that project.

I.e. there is no trivial way to simply throw that repo into Akash to run it there.

**Comment by Moekali84** [comment]
How do i SSH into my deployment after i have built it with something like putty?

**Reply by ghost** [reply]
By default you can always `lease-shell` into your deployment.
But since there is a known issue where this feature breaks after some time (when akash-provider service pod gets restarted), I recommend this workaround - basically it adds sshd service into your deployment so you can always ssh into your container pod.
And is generally more practical to have SSH access to your deployment.

The how-to add the sshd into your deployment is here:
https://github.com/akash-network/support/issues/87#issuecomment-1759421892

Please let me know if you have any further questions.
